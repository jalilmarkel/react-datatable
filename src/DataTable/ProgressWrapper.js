import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const ProgressWrapperStyle = styled.div`
  position: ${props => (props.centered ? 'absolute' : 'relative')};
  ${props => props.centered && 'align-items: center'};
  text-align: center;
  display: flex;
  justify-content: center;
  width: 100%;
  height: 100%;
`;

const ProgressWrapper = ({ component, centered }) => (
  <ProgressWrapperStyle centered={centered}>
    {component}
  </ProgressWrapperStyle>
);

ProgressWrapper.propTypes = {
  component: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.node,
    PropTypes.func,
  ]).isRequired,
  centered: PropTypes.bool,
};

ProgressWrapper.defaultProps = {
  centered: false,
};

export default ProgressWrapper;
