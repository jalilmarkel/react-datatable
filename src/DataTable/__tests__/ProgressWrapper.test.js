import 'jest-styled-components';
import React from 'react';
import { render, cleanup } from 'react-testing-library';

import ProgressWrapper from '../ProgressWrapper';

afterEach(cleanup);

test('component <ProgressWrapper component /> should render correctly when a component is passed that is a string', () => {
  const { container } = render(<ProgressWrapper component="A Component that is passed in">A String that is passed in</ProgressWrapper>);

  expect(container.firstChild).toMatchSnapshot();
});

test('component <ProgressWrapper component /> should render correctly when a component is passed that is a react component', () => {
  const { container } = render(<ProgressWrapper component={<div>A String that is passed in</div>}><div>A String that is passed in</div></ProgressWrapper>);

  expect(container.firstChild).toMatchSnapshot();
});

test('component <ProgressWrapper centered /> should render correctly when a component is passed that is a react component', () => {
  const { container } = render(<ProgressWrapper centered component="A Component that is passed in" />);

  expect(container.firstChild).toMatchSnapshot();
});
