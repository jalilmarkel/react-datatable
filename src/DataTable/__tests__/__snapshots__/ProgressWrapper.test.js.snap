// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component <ProgressWrapper centered /> should render correctly when a component is passed that is a react component 1`] = `
.c0 {
  position: absolute;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  height: 100%;
}

<div
  className="c0"
>
  A Component that is passed in
</div>
`;

exports[`component <ProgressWrapper component /> should render correctly when a component is passed that is a react component 1`] = `
.c0 {
  position: relative;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  height: 100%;
}

<div
  className="c0"
>
  <div>
    A String that is passed in
  </div>
</div>
`;

exports[`component <ProgressWrapper component /> should render correctly when a component is passed that is a string 1`] = `
.c0 {
  position: relative;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  height: 100%;
}

<div
  className="c0"
>
  A Component that is passed in
</div>
`;
